tiledb/query_condition.py

```
@@ -57,11 +58,27 @@ class QueryCondition(ast.NodeVisitor):
     def __init__(self, expression="", ctx=None):
         if ctx is None:
             ctx = tiledb.default_ctx()
+
         self._ctx = ctx
+        self._schema = None
+        self._query_attrs = None
+        self._c_obj = None
 
-        tree = ast.parse(expression)
+        self.tree = ast.parse(expression)
         self.raw_str = expression
-        self._c_obj = self.visit(tree.body[0]) if tree.body else qc.qc(self._ctx)
+
+    def init_query_condition(self, schema, query_attrs):
+        self._schema = schema
+        self._query_attrs = query_attrs
+
+        if self.tree.body:
+            self._c_obj = self.visit(self.tree.body[0])
+
+            if not isinstance(self._c_obj, tiledb._query_condition.qc):
+                raise tiledb.TileDBError(
+                    "Malformed query condition statement. A query condition must "
+                    "be made up of one or more Boolean expressions."
+                )
```

Do we need to raise an error in the (invisible) `else` case here?